lambda-s3/src/main/java/com/formkiq/stacks/lambda/s3/StagingS3Create.java

```diff
@@ -843,4 +851,22 @@ private void writeS3Document(final LambdaLogger logger, final String bucket, fin
           map);
     }
   }
+
+  @SuppressWarnings("unchecked")
+  private void handleCompressionRequest(final String bucket, final String key) {
+    final String contentString = this.s3.getContentAsString(bucket, key, null);
+    final Map<String, Object> content = this.gson.fromJson(contentString, Map.class);
+    final String siteId = content.get("siteId").toString();
+    final String archiveKey = key.replace(".json", ".zip");
+    Type jsonStringList = new TypeToken<List<String>>() {}.getType();
+    final List<String> documentIds =
+        gson.fromJson(content.get("documentIds").toString(), jsonStringList);
+
+    try {
+      this.documentCompressor.compressDocuments(siteId, this.documentsBucket, bucket, archiveKey,
+          documentIds);
+    } catch (Exception e) {
+      // TODO: Log
```

Comment:
exception should be thrown all the way up to the handler

Reply:
Done.